Update all of the docs to reflect new org service details (#244)
moderneinc/moderne-docs: docs/administrator-documentation/moderne-platform/how-to-guides/agent-configuration/org-service-migration.md

---
sidebar_label: Organizations service migration guide
description: How to migrate repos.csv from the org service to the agent.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Organizations service migration

> In order to simplify Moderne's operational complexity, we decided that configuring organizational hierarachies should only require creating a file and giving it to the Moderne agent (either via direct access or via an unauthenticated HTTP endpoint), rather than running a dedicated Organizations service.

Reviewer: ```suggestion
In order to simplify Moderne's operational complexity, we decided that configuring organizational hierachies should only require creating a file and giving it to the Moderne agent (either via direct access or via an unauthenticated HTTP/S endpoint), rather than running a dedicated Organizations service.
```

> In order to simplify Moderne's operational complexity, we decided that configuring organizational hierarchies should only require creating a file and giving it to the Moderne agent (either via direct access or via an unauthenticated HTTP/S endpoint), rather than running a dedicated Organizations service.

Author: I feel like the "we" demonstrates ownership and that we stand by it. "It was decided" sounds more like someone in the chain decided it and we're just following orders. Not saying one way is inherently right or not - but I don't feel the same in regards to the "we decided" tone.

While you can still [run an Organizations service](../org-service.md) to restrict access to various repositories/organizations or to customize commit messages by repository, the functionality around org hierarchies is being moved to the Agent.

This guide will walk you through everything you need to know to migrate this functionality to the Moderne Agent.

## Migration instructions

At a high-level, the migration process is as follows:

> 1. Remove any configuration related to the org hierarchies or DevCenters from your Agent

Reviewer: ```suggestion
1. Remove any outdated configuration related to the org hierarchies or DevCenters from your Agent
```

| Argument Name                                    |
|--------------------------------------------------|
| `--moderne.agent.organization.file.reposCsvPath` | 

> | `--moderne.agent.organization.commitOptionsPath` | 

Reviewer 2: do we need to highlight that commitOptions are no longer supported in org service? They are now global and available in agent only. Desjardins may complain. FYI @kmccarp @nadavc

Author: I'm down to make any changes based on the feedback but unsure what I should/shouldn't say right now.

</TabItem>
</Tabs>

### 2. Create the new files and put them somewhere the Agent can access

> 1. Copy the `devcenter.json` file from your Organization Service and put it somewhere where your Agent can access. This could mean putting this file on the same file system that Agent has access to – or it could mean putting it behind an unauthenticated HTTP endpoint.

Reviewer: ```suggestion
1. Copy the `devcenter.json` file from your Organization Service and put it somewhere where your Agent can access. This could mean putting this file on the same file system that Agent has access to – or it could mean putting it behind an unauthenticated HTTP/S endpoint.
```

> 2. Follow [our guide for creating a repos.csv file](../../../references/repos-csv.md). Alternatively, if your Organization Service already uses a `repos.csv`, you may copy that file directly and put it somewhere where your Agent can access (either by putting it on a file system the agent has access to or by putting the file behind an unauthenticated HTTP endpoint that the agent can call).

Reviewer: ```suggestion
2. Follow [our guide for creating a repos.csv file](../../../references/repos-csv.md). Alternatively, if your Organization Service already uses a `repos.csv`, you may copy that file directly and put it somewhere where your Agent can access (either by putting it on a file system the agent has access to or by putting the file behind an unauthenticated HTTP/S endpoint that the agent can call).
```

| Environment Variable                                       | Default | Description |
|------------------------------------------------------------|---------|-------------|
| `MODERNE_AGENT_ORGANIZATION_REPOSCSV`                      |         | The path of your `repos.csv` file that provides organization information. This could also be an unauthenticated HTTP/S URI in the form of `https://your-serve/repos.csv`. |

> | `MODERNE_AGENT_ORGANIZATION_DEVCENTER`                     |         | The path of your `devcenter.json` file that provides the DevCenter configurations. This could also be an unauthenticated HTTP/S URI in the form of `https://your-service/devcenter.json`. |

Reviewer: You have `your-serve` on the first line and `your-service` on the second, I assume you're trying to show that they can be different services, but maybe then you can do something like `https://<internal-endpoint>`

# ... Existing arguments
--moderne.agent.organization.reposCsv=/Users/MY_USER/Documents/repos.csv \
--moderne.agent.organization.devCenter=/Users/MY_USER/Documents/devcenter.json \
--moderne.agent.defaultCommitOptions[0]=PullRequest \
--moderne.agent.defaultCommitOptions[1]=ForkAndPullRequest \
--moderne.agent.organization.updateIntervalSeconds=600 \
# ... Additional arguments
```

</TabItem>
</Tabs>

:::tip
The Agent is able to fetch your `repos.csv` from an unauthenticated HTTP/HTTPS endpoint. To specify an endpoint instead
of a local file, use a URI instead of a local file.
:::

> ### 3. Remove the organization service from your system

Author: I don't understand this comment. What is missing? I see three steps and three numbers.

> ### 3. Remove the organization service from your system

Reviewer 3: Maybe my IDE is screwed up but I'm not seeing a header that starts with `### 4.` ?

> ### 3. Remove the organization service from your system

Author: I don't see why there would be a 4?
